backend/test/integrated/teamspace.js

```diff
@@ -366,4 +373,304 @@ describe("Checking Quota Info  ", function() {
+				.send({
+					riskCategories: duplicateRiskCategoryLabels
+				})
+				.expect(400, function(err, res) {
+					expect(res.body.value).to.equal(responseCodes.RISK_DUPLICATE_CATEGORY.value);
+					done(err);
+				});
+		});
+
+		it("with duplicate (case insensitive) topic type should fail", function(done) {
+			const duplicateTopicTypeLabels = ["clone 2", "CLONE 2"];
+			agent.put(`/${user.user}/settings`)
+				.send({
+					topicTypes: duplicateTopicTypeLabels
+				})
+				.expect(400, function(err, res) {
+					expect(res.body.value).to.equal(responseCodes.ISSUE_DUPLICATE_TOPIC_TYPE.value);
+					done(err);
+				});
+		});
+
+		after(function(done) {
+			this.timeout(timeout);
+			agent.post("/logout")
+				.expect(200, done);
+		});
+	});
+
+	describe("Get teamspace settings", function(done) {
+		const user =  imsharedTeamspace;
+
+		before(function(done) {
+			this.timeout(timeout);
+			agent.post("/login")
+				.send({username: user.user, password: user.password})
+				.expect(200, done);
+
+		});
+
+		it("should succeed", function(done) {
+			agent.get(`/${user.user}/settings`)
+				.expect(200, function(err, res) {
+					expect(res.body._id).to.equal(user.user);
+					expect(res.body.riskCategories).to.deep.equal(defaultRiskCategories);
+					expect(res.body.topicTypes).to.deep.equal(defaultTopicTypes);
+					done(err);
+				});
+		});
+
+		after(function(done) {
+			this.timeout(timeout);
+			agent.post("/logout")
+				.expect(200, done);
+		});
+	});
+
+	describe("Download mitigations file", function(done) {
+		const user =  imsharedTeamspace;
+
+		before(function(done) {
+			this.timeout(timeout);
+			agent.post("/login")
+				.send({username: user.user, password: user.password})
+				.expect(200, done);
+
+		});
+
+		it("that doesn't exist should fail", function(done) {
+			agent.get(`/${user.user}/settings/mitigations.csv`)
+				.expect(404, function(err, res) {
+					expect(res.body.value).to.equal(responseCodes.NO_FILE_FOUND.value);
+					done(err);
+				});
+		});
+
+		after(function(done) {
+			this.timeout(timeout);
+			agent.post("/logout")
+				.expect(200, done);
+		});
+	});
+
+	describe("Upload mitigations file", function(done) {
+		const user =  imsharedTeamspace;
+		const notMitigationsFile = "/../../statics/mitigations/notMitigations.zip";
+
+		before(function(done) {
+			this.timeout(timeout);
+			agent.post("/login")
+				.send({username: user.user, password: user.password})
+				.expect(200, done);
+
+		});
+
+		it("should succeed", function(done) {
+			agent.post(`/${user.user}/settings/mitigations.csv`)
+				.attach("file", __dirname + mitigationsFile)
+				.expect(200, done);
+		});
+
+		it("non-CSV file should fail", function(done) {
+			agent.post(`/${user.user}/settings/mitigations.csv`)
+				.attach("file", __dirname + notMitigationsFile)
+				.expect(400, function(err, res) {
+					expect(res.body.value).to.equal(responseCodes.FILE_FORMAT_NOT_SUPPORTED.value);
+					done(err);
+				});
+		});
```

Comment:
We also need to do quota check here and maximum file size check here, I remember you added in quota test in the code.